x/auth/types/account.go

```diff
@@ -120,38 +140,24 @@ func (acc BaseAccount) String() string {
 	return out.(string)
 }
 
-type baseAccountPretty struct {
-	Address       sdk.AccAddress `json:"address" yaml:"address"`
-	PubKey        string         `json:"public_key" yaml:"public_key"`
-	AccountNumber uint64         `json:"account_number" yaml:"account_number"`
-	Sequence      uint64         `json:"sequence" yaml:"sequence"`
-}
-
 // MarshalYAML returns the YAML representation of an account.
 func (acc BaseAccount) MarshalYAML() (interface{}, error) {
-	alias := baseAccountPretty{
-		Address:       acc.Address,
-		AccountNumber: acc.AccountNumber,
-		Sequence:      acc.Sequence,
-	}
-
-	if acc.PubKey != nil {
-		pks, err := sdk.Bech32ifyPubKey(sdk.Bech32PubKeyTypeAccPub, acc.GetPubKey())
-		if err != nil {
-			return nil, err
-		}
-
-		alias.PubKey = pks
-	}
-
-	bz, err := yaml.Marshal(alias)
+	bz, err := codec.MarshalYAML(codec.NewProtoCodec(codectypes.NewInterfaceRegistry()), &acc)
 	if err != nil {
 		return nil, err
 	}
-
 	return string(bz), err
```

Comment:
Marshalers return `[]byte` normally. Why do we convert to string?

Reply:
not sure why, removing it doesn't seem to cause any issue but I'd rather get other contributors's opinion on that first as it doesn't really relate to changes from this PR anyway.

Reply:
hmm, I'm also not sure why. All MarshalYAML in the codebase return string, so at least it's consistent. Might be a follow-up PR to switch all to bytes, but it's a low-priority cleanup.